src/Storage/Query/Handler/PagingHandler.php

```diff
@@ -21,17 +22,26 @@ public function __invoke(ContentQueryParser $contentQuery)
         $cleanSearchQuery = $contentQuery->getService('search');
 
         foreach ($contentQuery->getContentTypes() as $contenttype) {
+            $parameters = $contentQuery->getParameters();
+
             //Find out if we are searching or just doing a simple query
-            if ($contentQuery->hasParameter('filter')) {
+            if (array_key_exists('filter', $parameters)) {
                 $query = clone $cleanSearchQuery;
             } else {
                 $query = $contentQuery->getService('select');
             }
 
             $repo = $contentQuery->getEntityManager()->getRepository($contenttype);
-            $query->setQueryBuilder($repo->createQueryBuilder('_' . $contenttype));
+
+            if (Version::compare('3.3', '<=')) {
```

Comment:
@GawainLynch There was a piece of code you made, that was:
```
if (version_compare(BoltVersion::forComposer(), '3.3.0', '<')) {
```
and I got confused.

But it works, so nothing to see here 😆